### PR TITLE
ci(github-action): periodically parse oss airflow issue to check provider release

### PR DESCRIPTION
**Pull request by Lee-W**
Check whether the RC testing announcement (e.g., https://github.com/apache/airflow/issues/31322) has been made on [apache-airflow](https://github.com/apache/airflow/). If there's an announcement and we do not have an existing open issue with the title `[DO NOT MERGE] Test RC provider packages for $rc_issue_url`, then trigger the rest of the RC provider testing workflow

### CODE REVIEW

**Comment by Lee-W** [comment]
@utkarsharma2 this is the wip commit i mentioned earlier today

**Comment by Lee-W** [comment]
@utkarsharma2 I just wrapped it up after working on issue-airflow. Please take a look if you're interested in it 🙂

File: .github/workflows/test-rc-release.yaml
Comment (utkarsharma2) [inline .github/workflows/test-rc-release.yaml:49]: @Lee-W Thanks for looking into this. 
I feel this is a bit risky since the value of `Status of testing Providers that were prepared on` might change from ticket to ticket and we will miss the testing of RC. WDYT?

Comment (utkarsharma2) [reply .github/workflows/test-rc-release.yaml:49]: I can think of a couple of solutions like sentiment analysis, but I think they are overkill for this situation.

Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:49]: Yes, such a check might be risky. I just took a look at the [airflow repo](https://github.com/apache/airflow/issues?q=is%3Aissue+%22Status+of+testing+Providers+that+were+prepared+on%22). There are around 50 issues in this format, so I'd say this is somewhat reliable. Even sentiment analysis might have some risk. Considering this, I think this might be a good enough solution, and we can always trigger manually if we notice this is not triggered.

Comment (uranusjr) [reply .github/workflows/test-rc-release.yaml:49]: The issue is actually generated from a script, so the title should be fairly predicatable:

https://github.com/apache/airflow/blob/8dc895cbb64e991893705b04e03a6655c35888fe/dev/breeze/src/airflow_breeze/commands/release_candidate_command.py#L248

If this still seems too unstructured, the *testing status* label can also be checked; I think the label is only used for this particular purpose right now.

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:39]:
```suggestion
      - name: Parse the latest 100 GitHub issues from apache-airflow to check providers testing announcement
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:59]: Shouldn't we check for PRs instead of issues here? Also, where are we checking out the `astronomer-providers` repo for checking the existing PRs in our repo?

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:113]: Shouldn't we also add a check here that there is no existing PR in our repo?

**Review by pankajkoti — COMMENTED** [review]
Looks great to me. Have some questions inline.

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:59]: In the current design, I'd say checking issues might make more sense as we use the timestamp as part of the branch name. But I'm thinking of renaming it as `rc-test-for-issue-$issue_number`

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:113]: if we have existing PR, we'll set this value to ''. https://github.com/astronomer/astronomer-providers/pull/1146/files#diff-7cd74bfb9babddfbfa7911a72e5d0784f450360d0a81e9f8319ecba30711b059R88

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:59]: If we all agree on this change, I think I'll be able to work on it this week. Or we can merge this one and create a issue for this improvement

Comment (pankajkoti) [reply .github/workflows/test-rc-release.yaml:59]: maybe I missing something or not understanding. Are we planning to make a change with the automation that along with PRs we will also create issues in astronomer-providers for the RCs? If yes, how are we planning to create such issues?

Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:59]: Ah, it's me who misunderstood the workflow 🤦‍♂️ We actually create a PR with that title instead of the issue. Thanks for finding it out! I'll fix it late today

Comment (pankajkoti) [reply .github/workflows/test-rc-release.yaml:59]: okay, sorry for the late review on this one

Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml:59]: Just fixed it. Could you pelase check again when you have time? Thanks!

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:59]:
```suggestion
      - name: Parse the latest GitHub pull requests for checking existing RC provider testing pull request
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:65]:
```suggestion
          testing_pr_url=`gh pr list \
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:77]:
```suggestion
          echo "testing_pr_url=$testing_issue_url" >> $GITHUB_OUTPUT
```

Comment (pankajkoti) [reply .github/workflows/test-rc-release.yaml:77]: and need to update the references in subsequent steps.

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:85]:
```suggestion
            echo "No RC providers testing announcement found on apache-airflow"
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:30]:
```suggestion
  check-rc-testing-announcement:
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:39]:
```suggestion
      - name: Parse the latest 100 GitHub issues from apache-airflow to check providers testing announcement
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:107]: This needs to be an `or` condition right? validate-manual-input runs only on `workflow_dispatch` whereas `check-rc-testing-annocement` runs only on `schedule`, so both will not be true at the same time and this job will never be executed then, no?

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:63]:
```suggestion
          # The default limit is 30. Set it to 100 for retrieving more pull requests
```

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [inline .github/workflows/test-rc-release.yaml:65]: I think this only lists open PRs. Once will close the PRs, and since the issue could still be there in apache-airflow in top 100 issues, are we at risk of recreating PRs? I think we should list here merged PRs too

**Review by pankajkoti — COMMENTED** [review]
Will it be possible to test this workflow somehow?

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml]: AFAIK, we don't have a way to do it in `needs`. That's why I use `always()` [here](https://github.com/astronomer/astronomer-providers/pull/1146/files#diff-7cd74bfb9babddfbfa7911a72e5d0784f450360d0a81e9f8319ecba30711b059R110)

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml]: Yes, that makes sense. I waited for the airflow issue to close before I closes the PR in the previous releases. Let me add it 💪 Thanks!

Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml]: just added

**Comment by Lee-W** [comment]
> Will it be possible to test this workflow somehow?

Yes, we can simulate through the latest closed issue (we will need to change the code to read closed issue from airflow for testing).

File: .github/workflows/test-rc-release.yaml
Comment (pankajkoti) [reply .github/workflows/test-rc-release.yaml]: okay, so just for my understanding, we are guaranteed that this job will run, right? Although `validate-manual-input` or `check-rc-testing-annocement` is skipped based on their if conditions, since we have always, this job will be run, correct? If that is correct, yes, please mark this conversation as resolved. Sorry for my lack of expertise here :)

**Review by pankajkoti — APPROVED** [review]
Looks good to me. IMO, we can merge this if we want the code to be available in main to test something. We can always fix something later if something is not working well.

File: .github/workflows/test-rc-release.yaml
Comment (Lee-W) [reply .github/workflows/test-rc-release.yaml]: Yes, that's how it works. Thanks for looking so deep into it 🙌 It also took me some time to sort all these stuff out

**Comment by Lee-W** [comment]
@pankajkoti We don't actually need it to be in the main branch for testing, but I think it is better to merge it to the main branch earlier. I'll do some simple tests on another branch to verify the basic functionality again (as we have made some changes these days). Once those work as I expect, I'll merge this one. Thanks for reviewing 👍

**Pull request merged**
